### PR TITLE
[feature fix][#2537]Fix Google Drive's folder names in project setting… 

### PR DESCRIPTION
**Pull request by hmoco**
…s tab not showing special characters. 
https://github.com/CenterForOpenScience/osf.io/issues/2537

## Purpose:

Google Drive's folders and files use URL encoding for their names. This can be properly displayed by using the standard method decodeURIComponent(), which has been previously used in the log history and file viewer code to regulate display of Google Drive parts. However, on the project setting page, where addon folders can be selected, all Google Drive names appear encoded (the functional side effect is that special characters are not shown properly, e.g space appears as %20). This fix creates exceptions on the folder picker for Google Drive folder so that decodeURIComponent() is called on their names before being displayed.

## Changes:
Google Drive's folder and file names now appear URL decoded universally in the OSF. Example:
Google Drive's folder names

## Side Effects:
Anything that uses folder picker.js to display folders should now show folder names correctly for Google Drive. There is a possibility this could be handled somewhere else, and that exceptions should not have been written on the main folder picker file for an addon.

### CODE REVIEW

File: website/static/js/addonNodeConfig.js [hunk 1]
```diff
@@ -39,7 +39,13 @@ var AddonFolderPickerViewModel = oop.extend(FolderPickerViewModel, {
             return (userHasAuth && selected) ? selected.type : '';
         });
         self.messages.submitSettingsSuccess =  ko.pureComputed(function() {
-            return 'Successfully linked "' + $osf.htmlEscape(self.folder().name) + '". Go to the <a href="' +
+            if(addonName === "Google Drive"){
```
Comment (samchrisinger) [inline website/static/js/addonNodeConfig.js]: Is there any reason not to use ```self.options.decodeFolder``` here?

Comment (hmoco) [inline website/static/js/addonNodeConfig.js]: Just did!

File: website/static/js/folderpicker.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ function treebeardTitleColumn(item, col) {
     return m('span', {
         className: cls,
         onclick: onclick
-    }, item.data.name);
+    }, tb.options.decodeFolder(item.data.name));
```
Comment (samchrisinger) [inline website/static/js/folderpicker.js]: :+1: nice, thanks for doing this.

**Comment by samchrisinger** [comment]
Please look at the JS tests in webite/static/js/tests/addons/, and add test cases for the functions you've modified.

**Comment by hmoco** [comment]
@samanehsan I think this might be good to go, removed the WIP tag.

**Comment by samchrisinger** [comment]
With the pushed changes, I think this PR looks good.

**Pull request merged**